pimd/pim_nb_config.c

```diff
@@ -114,6 +115,58 @@ static void pim_if_membership_refresh(struct interface *ifp)
 	pim_ifchannel_delete_on_noinfo(ifp);
 }
 
+/*
+ * pim_if_igmp_version_change is called when the igmp version changes
+ * If the igmp version is changed from 3 to 2, delete the 224.0.0.2
+ * membership from the socket so that it won't accept any packets.
+ * If the version is changed 2 to 3, add the membership to the socket.
+ */
+static void pim_if_igmp_version_change(struct interface *ifp)
+{
+	int ret = -1;
+	struct in_addr group;
+	struct listnode *sock_node;
+	struct gm_sock *igmp;
+	struct pim_interface *pim_ifp;
+	struct in_addr ifaddr;
+
+	pim_ifp = ifp->info;
+	ifaddr = pim_ifp->primary_address;
+
+	/* scan igmp sockets */
+	for (ALL_LIST_ELEMENTS_RO(pim_ifp->igmp_socket_list, sock_node, igmp)) {
+		if (inet_aton(PIM_ALL_IGMP_ROUTERS, &group)) {
```

Comment:
Is there any particular reason this function is converting PIM_ALL_IGMP_ROUTERS to a `struct in_addr` every time we are in the loop?

Reply:
Fixed